File: _includes/snippets/java/SimpleExample.java
class State {
    private List<String> _alreadyGreeted = new LinkedList<>();

    public static State create() {
        return new State();
    }

    String greetings(String name){
        String greeting = "Hello " + name + "\nI already greeted all these guys: " + _alreadyGreeted;
        _alreadyGreeted.add(name);
        return greeting;
    }
}

File: _includes/snippets/rust/simple-example.rs
struct State {
    already_greeted: Vec<String>
}

impl State {
    pub fn new() -> State {
        State {
            already_greeted : Vec::new()
        }
    }

    fn greetings(&mut self, name:String) -> String {
        let ag = self.already_greeted.join(", ");
        let greeting = format!("Hello {}\nI already greeted all these guys: [{}]\n", name, ag);
        self.already_greeted.push(name);
        greeting
    }
}
